### PR TITLE
Documented /oauth/

### PR DESCRIPTION
**Pull request by harshkhandeparkar**

### CODE REVIEW

File: controllers/oauth.go [hunk 1]
```diff
@@ -24,13 +24,30 @@ type OAuthResBodyFields struct {
...
+//
+//	@Summary		OAuth Authentication
+//	@Description	Authenticates a user throught Github OAuth.
+//	@Accept			json
+//	@Produce		json
+//	@Param			request	body		OAuthReqBodyFields	true	"Fields required for authentication."
+//	@Success		200		{object}	OAuthResBodyFields	"Succesfully authenticated."
+//	@Failure		400		{object}	utils.HTTPMessage	"Error decoding JSON body."
+//	@Failure		400		{object}	utils.HTTPMessage	"Empty body parameters."
+//	@Failure		500		{object}	utils.HTTPMessage	"Error getting OAuth access token."
```
Comment (YoganshSharma) [inline controllers/oauth.go:45]: only the last `@Failure  500` annotation is considered

**Review by YoganshSharma — CHANGES_REQUESTED** [review]

File: controllers/oauth.go (hunk 1 above)
Comment (harshkhandeparkar) [reply controllers/oauth.go:45]: Is it not possible to add multiple?

Comment (YoganshSharma) [reply controllers/oauth.go:45]: The error code is used as a key, so no

Comment (YoganshSharma) [reply controllers/oauth.go:45]: I can see two options here
- document this in the description tag of the API documenting all the possible behaviours (btw you can add [description over multiple lines](https://github.com/swaggo/swag#descriptions-over-multiple-lines))
- Or, write a long description for the failure tag .

Comment (harshkhandeparkar) [reply controllers/oauth.go:45]: hmm

Comment (harshkhandeparkar) [reply controllers/oauth.go:45]: I'll choose the second option

Comment (harshkhandeparkar) [reply controllers/oauth.go]: Done, please review.

**Review by YoganshSharma — CHANGES_REQUESTED** [review]

**Review by YoganshSharma — APPROVED** [review]
LGTM

**Pull request merged**